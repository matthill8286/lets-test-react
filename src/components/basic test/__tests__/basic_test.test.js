import React from 'react';
import Basic from '../basic_test';

import Enzyme, { shallow, mount } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

// Basic Test with React-test-renderer
it('renders correctly react-test-renderer', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Basic />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});


// Basic Test with Enzyme
it('renders correctly shallow enzyme', () => {
  const wrapper = shallow(<Basic />)

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders correctly mount enzyme', () => {
  const wrapper = mount(<Basic />)

  expect(toJson(wrapper)).toMatchSnapshot();
});
