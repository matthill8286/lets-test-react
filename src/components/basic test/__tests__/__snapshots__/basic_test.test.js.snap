// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly mount enzyme 1`] = `
<Basic>
  <div>
    <h1>
       Basic Test
    </h1>
    <p>
       This is a basic Test Component
    </p>
  </div>
</Basic>
`;

exports[`renders correctly react-test-renderer 1`] = `
<div>
  <h1>
     Basic Test
  </h1>
  <p>
     This is a basic Test Component
  </p>
</div>
`;

exports[`renders correctly shallow enzyme 1`] = `
<div>
  <h1>
     Basic Test
  </h1>
  <p>
     This is a basic Test Component
  </p>
</div>
`;
